gz_ros2_control/include/gz_ros2_control/gz_ros2_control_plugin.hpp

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GZ_ROS2_CONTROL__GZ_ROS2_CONTROL_PLUGIN_HPP_
+#define GZ_ROS2_CONTROL__GZ_ROS2_CONTROL_PLUGIN_HPP_
+
+#include <memory>
+
+#ifdef GZ_HEADERS
+#include <gz/sim/System.hh>
+#define GZ_SIM_NAMESPACE gz::sim::
+#else
+#include <ignition/gazebo/System.hh>
+#define GZ_SIM_NAMESPACE ignition::gazebo::
+#endif
```

Comment:
I recommend doing using namespace aliases instead of macros here. So you'd have `namespace sim = gz::sim` or `namespace sim = ignition::gazebo` based on `GZ_HEADERS` and then using `sim` as a prefix for all the entities (eg. `sim::System`, `sim::Entity`, etc).